File: DSA Practice/LeetCode/1572. Matrix Diagonal Sum/solution.java
class Solution {
    public int diagonalSum(int[][] mat) {
        int s = 0;
        int f = 0;
        int b = mat[0].length - 1;

        for (int[] i : mat){
            s += i[f];
            i[f] = 0;

            s += i[b];
            i[b] = 0;

            f++;
            b--;
        }

        return s;
    }
}

File: DSA Practice/LeetCode/1572. Matrix Diagonal Sum/solution.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        s = 0
        f = 0
        b = len(mat[0]) - 1
        for i in mat:
            #print(i[f], i[b])
            s += i[f]
            i[f] = 0
            s += i[b]
            i[b] = 0

            f += 1
            b -= 1

        return s